### PR TITLE
Update 020_ros_with_dds.md

### PR DESCRIPTION
**Pull request by RamziKaroui**
Add a reference to the new Tier-1 Cyclone DDS implementation and added some insights on OpenspliceDDS implementation

### CODE REVIEW

File: articles/020_ros_with_dds.md [hunk 1]
```diff
@@ -114,7 +114,7 @@ From RTI's website ([http://community.rti.com/kb/xml-qos-example-using-rti-conne
 > In some circumstances, the TCP protocol might be needed for discovery and data exchange.
 > For more information on the RTI TCP Transport, please refer to the section in the RTI Core Libraries and Utilities User Manual titled "RTI TCP Transport".
 
-From ADLINK's website, they support TCP as of OpenSplice v6.4:
+From ADLINK's website, they support TCP Since OpenSplice v6.4, supports shared memory as a transport and as an in-memory database.
```
Comment (gbiggs) [inline articles/020_ros_with_dds.md]:
```suggestion
From ADLINK's website, they support TCP as of OpenSplice v6.4, supports shared memory as a transport and as an in-memory database.
```

Comment (gbiggs) [reply articles/020_ros_with_dds.md]: What version do they support shared memory as a transport from, and what version do they support an in-memory database from?

File: articles/020_ros_with_dds.md [hunk 2]
```diff
@@ -151,6 +151,8 @@ It is available on [Github](https://github.com):
 ADLINK's implementation comes with a basic, functioning build system and was fairly easy to package.
 OpenSplice appears to be the number two DDS implementation in use, but that is hard to tell for sure.
 
+ADLINK also supports heavily the Eclipse Cyclone DDS implementation [https://github.com/eclipse-cyclonedds/cyclonedds] (https://www.adlinktech.com/en/CycloneDDS).  Cyclone DDS is licensed under Eclipse Public License - v 2.0 ( https://www.eclipse.org/legal/epl-2.0/).
```
Comment (gbiggs) [inline articles/020_ros_with_dds.md]:
```suggestion
ADLINK also contributes to Eclipse Cyclone DDS [https://github.com/eclipse-cyclonedds/cyclonedds] (https://www.adlinktech.com/en/CycloneDDS).  Cyclone DDS is licensed under the Eclipse Public License - v 2.0 ( https://www.eclipse.org/legal/epl-2.0/).
```

File: articles/020_ros_with_dds.md [hunk 3]
```diff
@@ -163,7 +165,7 @@ eProsima's implementation also includes a minimum DDS API, IDL support, and auto
 
 Given the relatively strong LGPL option and the encouraging but custom license from RTI, it seems that depending on and even distributing DDS as a dependency should be straightforward.
 One of the goals of this proposal would be to make ROS 2 DDS vendor agnostic.
-So, just as an example, if the default implementation is Connext, but someone wants to use one of the LGPL options like OpenSplice or FastRTPS, they simply need to recompile the ROS source code with some options flipped and they can use the implementation of their choice.
+So, just as an example, if the default implementation is Connext, but someone wants to use one of the LGPL options like OpenSplice, FastRTPS or Cyclone DDS, they simply need to recompile the ROS source code with some options flipped and they can use the implementation of their choice.
```
Comment (gbiggs) [inline articles/020_ros_with_dds.md]: Cyclone DDS is not LGPL. Actually, neither is Fast DDS...

```suggestion
So, just as an example, if the default implementation is Connext, but someone wants to use one of the Apache 2 options like Fast DDS or Cyclone DDS, they simply need to recompile the ROS source code with some options flipped and they can use the implementation of their choice.
```

**Review by gbiggs — CHANGES_REQUESTED** [review]

File: articles/020_ros_with_dds.md (hunk 1 above)
Comment (RamziKaroui) [reply articles/020_ros_with_dds.md]: OpenSpliceDDS ( initially called SPLICE) was actually built from the begging as in-memory eventually consistent database, See slides 8 of Erik Boasson on the different spirits that  were the DDS technology ( https://static1.squarespace.com/static/51df34b1e4b08840dcfd2841/t/5d102784919bd40001a4059d/1561339788292/2_3_Boasson.pdf ) 
More details on this link
http://download.prismtech.com/docs/Vortex/html/ospl/DeploymentGuide/overview.html#shared-memory-architecture  

As a shared memory, it is used for inter-process communication and therefore as a local transport layer.

Comment (dirk-thomas) [reply articles/020_ros_with_dds.md]: Isn't the shared memory transport only available in the commericial version but not in the community edition?

File: articles/020_ros_with_dds.md [hunk 4]
```diff
@@ -151,6 +151,8 @@ It is available on [Github](https://github.com):
 ADLINK's implementation comes with a basic, functioning build system and was fairly easy to package.
 OpenSplice appears to be the number two DDS implementation in use, but that is hard to tell for sure.
 
+ADLINK also contributes to Eclipse Cyclone DDS [https://github.com/eclipse-cyclonedds/cyclonedds] (https://www.adlinktech.com/en/CycloneDDS).  Cyclone DDS is licensed under the Eclipse Public License - v 2.0 ( https://www.eclipse.org/legal/epl-2.0/).
```
Comment (dirk-thomas) [inline articles/020_ros_with_dds.md:154]: Nitpick: please start each sentence on a new line (following the style guide). Also remove the extra space after the open parenthesis.

File: articles/020_ros_with_dds.md [hunk 5]
```diff
@@ -163,7 +165,7 @@ eProsima's implementation also includes a minimum DDS API, IDL support, and auto
 
 Given the relatively strong LGPL option and the encouraging but custom license from RTI, it seems that depending on and even distributing DDS as a dependency should be straightforward.
 One of the goals of this proposal would be to make ROS 2 DDS vendor agnostic.
-So, just as an example, if the default implementation is Connext, but someone wants to use one of the LGPL options like OpenSplice or FastRTPS, they simply need to recompile the ROS source code with some options flipped and they can use the implementation of their choice.
+So, just as an example, if the default implementation is Connext, but someone wants to use one of the Apache 2 options like Fast DDS or Cyclone DDS, they simply need to recompile the ROS source code with some options flipped and they can use the implementation of their choice.
```
Comment (dirk-thomas) [inline articles/020_ros_with_dds.md:168]: While updating the license and changing the vendor names I would suggest to also update the surrounding information. Connext isn't the default - Fast DDS is. "Flipping some options" isn't really helpful without mentioning them - and afaik there shouldn't be any need for options - just rebuilding from source with the desired RMW implementation present should make it usable.

File: articles/020_ros_with_dds.md (hunk 1 above)
Comment (RamziKaroui) [reply articles/020_ros_with_dds.md]: From Version 6  shared memory is provided with LGPL licensing for Commercial Edition. If the intension is to only scope the Apache license V2.0 of the Opensplice Community version, let's make that clear in the document.

Comment (dirk-thomas) [reply articles/020_ros_with_dds.md]: The information doesn't have to be limited to the community version. But if the feature is only available in the commercial edition than that should be mentioned explicitly.

File: articles/020_ros_with_dds.md (hunk 4 above)
Comment (gbiggs) [inline articles/020_ros_with_dds.md:154]:
```suggestion
ADLINK also contributes to [Eclipse Cyclone DDS](https://github.com/eclipse-cyclonedds/cyclonedds) (see https://www.adlinktech.com/en/CycloneDDS).
Cyclone DDS is licensed under the [Eclipse Public License - v 2.0](https://www.eclipse.org/legal/epl-2.0/).
```

File: articles/020_ros_with_dds.md (hunk 5 above)
Comment (gbiggs) [inline articles/020_ros_with_dds.md:168]:
```suggestion
So, just as an example, if the default implementation is Fast DDS, but someone wants to use a commercial implementation like Connext, they simply need to change the `RMW_IMPLEMENTATION` environment variable and relaunch their nodes to change the underlying DDS implementation to Connext.
```

File: articles/020_ros_with_dds.md (hunk 1 above)
Comment (RamziKaroui) [reply articles/020_ros_with_dds.md]: That makes sense. The feature is available in the commercial edition.

**Comment by dirk-thomas** [comment]
@RamziKaroui Friendly ping.

**Comment by clalancette** [comment]
There's been no response to this for quite a while, so I'm going to go ahead and close it out.  Feel free to reopen and fix the review feedback.

**Pull request closed without merging**